### PR TITLE
Increase binary size limit and publish binary size data even if limit is exceeded.

### PR DESCRIPTION
**Pull request by edgchen1**
**Description**:
Increase binary size limit to fix "Linux CPU Minimal Build E2E CI Pipeline" in master.
Also updated the build definition to publish binary size data even if the size limit check fails so size data doesn't get lost.

**Motivation and Context**
Fix build.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/linux-cpu-minimal-build-ci-pipeline.yml [hunk 1]
```diff
@@ -179,15 +179,23 @@ jobs:
 
 
   - task: AzureCLI@2
-    displayName: 'Azure CLI'
-    condition: and (succeeded(), in(variables['Build.Reason'], 'IndividualCI', 'BatchedCI'), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+    displayName: 'Publish binary size data'
+    condition: and(succeededOrFailed(), in(variables['Build.Reason'], 'IndividualCI', 'BatchedCI'), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/linux-cpu-minimal-build-ci-pipeline.yml:183]: What if the build failed, will this still be executed?

Comment (gwang-msft) [reply tools/ci_build/github/azure-pipelines/linux-cpu-minimal-build-ci-pipeline.yml:183]: If build failed (not failed at where the binary size is compared), then the `binary_size_data.txt` will not be generated, so this step will fail also

**Review by gwang-msft — APPROVED** [review]

**Review by snnn — APPROVED** [review]

**Comment by snnn** [comment]
Edward is OOF today. Now I will merge this PR for him.

**Comment by snnn** [comment]
Do you know what caused the increase?

**Comment by edgchen1** [comment]
> Do you know what caused the increase?

Recently there have been several PRs which increased it. I think the size limit was exceeded by the combination of their changes in master.

**Pull request merged**